api/src/main/resources/deployment_manager.py

```diff
@@ -279,15 +283,22 @@ def _assert_application_status(self, application, required_status):
             else:
                 raise ConflictingState(json.dumps({'status': status}))
 
+        if status != ApplicationState.NOTCREATED:
+            created_user = self._application_registrar.get_application_user(application)
```

Comment:
I don't think an additional DB read is needed here as the information is already in app_info returned by get_application() isn't it?